File: salesflow-ai/src/types/vertical.ts
/**
 * Vertical Architecture - Shared Types
 * 
 * Definiert die JSON-Struktur für Vertical-Configs
 * Wird sowohl im Frontend (TypeScript) als auch Backend (Pydantic) verwendet
 */

/**
 * Feature-Flags für verschiedene Funktionsbereiche
 */
export interface VerticalFeatures {
  crm: boolean;
  finance: boolean;
  gamification: boolean;
  team: boolean;
  analytics: boolean;
  ai_coach: boolean;
  lead_hunter: boolean;
  follow_ups: boolean;
  templates: boolean;
  autopilot: boolean;
  cold_call: boolean;
  closing_coach: boolean;
  performance_insights: boolean;
  commission_tracker: boolean;
  genealogy: boolean;
  power_hour: boolean;
  churn_radar: boolean;
  roleplay_dojo: boolean;
  network_graph: boolean;
  field_ops: boolean;
  route_planner: boolean;
}

/**
 * Terminologie-Mapping für branchenspezifische Begriffe
 */
export interface VerticalTerminology {
  lead: string;
  deal: string;
  revenue: string;
  commission: string;
  prospect: string;
  customer: string;
  contact: string;
  pipeline: string;
  closing: string;
  sales: string;
  team: string;
  downline: string;
  partner: string;
  enrollment: string;
  signup: string;
  [key: string]: string; // Erlaubt zusätzliche benutzerdefinierte Begriffe
}

/**
 * AI-Kontext für branchenspezifische Prompts
 */
export interface VerticalAIContext {
  persona: string; // Beschreibung der AI-Persona
  focus_topics: string[]; // Themen, auf die sich die AI konzentrieren soll
  industry_terms: string[]; // Branchenspezifische Begriffe
  tone: string; // Kommunikationston (z.B. "professionell", "freundlich", "motivierend")
  examples: string[]; // Beispiel-Interaktionen
  avoid_topics: string[]; // Themen, die vermieden werden sollen
}

/**
 * Route-Konfiguration
 */
export interface VerticalRoutes {
  hidden: string[]; // Routes, die ausgeblendet werden sollen
  priority: string[]; // Routes, die priorisiert werden sollen
  custom_labels?: Record<string, string>; // Custom Labels für Routes
}

/**
 * Vollständige Vertical-Config
 */
export interface VerticalConfig {
  features: VerticalFeatures;
  terminology: VerticalTerminology;
  ai_context: VerticalAIContext;
  routes: VerticalRoutes;
}

/**
 * Vertical-Metadaten
 */
export interface Vertical {
  id: string;
  key: string; // z.B. "mlm", "real_estate", "finance"
  name: string; // Anzeigename
  description?: string;
  config: VerticalConfig;
  created_at?: string;
  updated_at?: string;
}

/**
 * Default MLM Config (Fallback)
 */
export const DEFAULT_MLM_CONFIG: VerticalConfig = {
  features: {
    crm: true,
    finance: true,
    gamification: true,
    team: true,
    analytics: true,
    ai_coach: true,
    lead_hunter: true,
    follow_ups: true,
    templates: true,
    autopilot: true,
    cold_call: true,
    closing_coach: true,
    performance_insights: true,
    commission_tracker: true,
    genealogy: true,
    power_hour: true,
    churn_radar: true,
    roleplay_dojo: true,
    network_graph: true,
    field_ops: false,
    route_planner: false,
  },
  terminology: {
    lead: "Lead",
    deal: "Einschreiben",
    revenue: "Umsatz",
    commission: "Provision",
    prospect: "Interessent",
    customer: "Partner",
    contact: "Kontakt",
    pipeline: "Pipeline",
    closing: "Abschluss",
    sales: "Verkauf",
    team: "Team",
    downline: "Downline",
    partner: "Partner",
    enrollment: "Einschreibung",
    signup: "Anmeldung",
  },
  ai_context: {
    persona: "Du bist ein erfahrener Network Marketing Coach, der motivierend und unterstützend kommuniziert.",
    focus_topics: [
      "Network Marketing",
      "Team-Aufbau",
      "Downline-Management",
      "Provisionen",
      "Motivation",
      "Einschreibungen",
    ],
    industry_terms: [
      "Einschreibung",
      "Downline",
      "Upline",
      "Provision",
      "PV",
      "GV",
      "Partner",
      "Team",
    ],
    tone: "motivierend",
    examples: [
      "Wie kann ich mein Team besser motivieren?",
      "Was sind die besten Strategien für Einschreibungen?",
    ],
    avoid_topics: ["Pyramid Scheme", "Betrug"],
  },
  routes: {
    hidden: [],
    priority: ["/dashboard", "/leads", "/team", "/genealogy"],
  },
};

/**
 * Default Real Estate Config (Beispiel)
 */
export const DEFAULT_REAL_ESTATE_CONFIG: VerticalConfig = {
  features: {
    crm: true,
    finance: true,
    gamification: false,
    team: false,
    analytics: true,
    ai_coach: true,
    lead_hunter: true,
    follow_ups: true,
    templates: true,
    autopilot: true,
    cold_call: true,
    closing_coach: true,
    performance_insights: true,
    commission_tracker: true,
    genealogy: false,
    power_hour: false,
    churn_radar: false,
    roleplay_dojo: true,
    network_graph: false,
    field_ops: true,
    route_planner: true,
  },
  terminology: {
    lead: "Interessent",
    deal: "Abschluss",
    revenue: "Umsatz",
    commission: "Provision",
    prospect: "Käufer",
    customer: "Kunde",
    contact: "Kontakt",
    pipeline: "Pipeline",
    closing: "Abschluss",
    sales: "Verkauf",
    team: "Team",
    downline: "",
    partner: "Partner",
    enrollment: "",
    signup: "Anmeldung",
  },
  ai_context: {
    persona: "Du bist ein professioneller Immobilienmakler mit langjähriger Erfahrung.",
    focus_topics: [
      "Immobilienverkauf",
      "Kundenberatung",
      "Objektpräsentation",
      "Verhandlungen",
      "Abschlüsse",
    ],
    industry_terms: [
      "Immobilie",
      "Objekt",
      "Käufer",
      "Verkäufer",
      "Maklerprovision",
      "Besichtigung",
    ],
    tone: "professionell",
    examples: [
      "Wie präsentiere ich eine Immobilie am besten?",
      "Wie verhandle ich erfolgreich einen Abschluss?",
    ],
    avoid_topics: [],
  },
  routes: {
    hidden: ["/genealogy", "/power-hour", "/churn-radar"],
    priority: ["/dashboard", "/leads", "/pipeline", "/field-ops"],
  },
};

File: backend/app/schemas/vertical.py
"""
Vertical Architecture - Pydantic Schemas

Definiert die JSON-Struktur für Vertical-Configs im Backend.
Spiegelt die TypeScript-Types aus src/types/vertical.ts.
"""

from __future__ import annotations

from typing import Dict, List, Optional

from pydantic import BaseModel, ConfigDict, Field


class VerticalFeatures(BaseModel):
    """Feature-Flags für verschiedene Funktionsbereiche."""

    model_config = ConfigDict(protected_namespaces=())

    crm: bool = True
    finance: bool = True
    gamification: bool = True
    team: bool = True
    analytics: bool = True
    ai_coach: bool = True
    lead_hunter: bool = True
    follow_ups: bool = True
    templates: bool = True
    autopilot: bool = True
    cold_call: bool = True
    closing_coach: bool = True
    performance_insights: bool = True
    commission_tracker: bool = True
    genealogy: bool = True
    power_hour: bool = True
    churn_radar: bool = True
    roleplay_dojo: bool = True
    network_graph: bool = True
    field_ops: bool = False
    route_planner: bool = False


class VerticalTerminology(BaseModel):
    """Terminologie-Mapping für branchenspezifische Begriffe."""

    lead: str = "Lead"
    deal: str = "Deal"
    revenue: str = "Umsatz"
    commission: str = "Provision"
    prospect: str = "Interessent"
    customer: str = "Kunde"
    contact: str = "Kontakt"
    pipeline: str = "Pipeline"
    closing: str = "Abschluss"
    sales: str = "Verkauf"
    team: str = "Team"
    downline: str = "Downline"
    partner: str = "Partner"
    enrollment: str = "Einschreibung"
    signup: str = "Anmeldung"

    model_config = ConfigDict(
        extra="allow",  # Erlaubt zusätzliche benutzerdefinierte Begriffe
        protected_namespaces=(),  # Unterdrückt Warnungen für "model_" Felder
    )


class VerticalAIContext(BaseModel):
    """AI-Kontext für branchenspezifische Prompts."""

    model_config = ConfigDict(protected_namespaces=())

    persona: str = Field(
        ...,
        description="Beschreibung der AI-Persona",
    )
    focus_topics: List[str] = Field(
        default_factory=list,
        description="Themen, auf die sich die AI konzentrieren soll",
    )
    industry_terms: List[str] = Field(
        default_factory=list,
        description="Branchenspezifische Begriffe",
    )
    tone: str = Field(
        default="professionell",
        description="Kommunikationston (z.B. 'professionell', 'freundlich', 'motivierend')",
    )
    examples: List[str] = Field(
        default_factory=list,
        description="Beispiel-Interaktionen",
    )
    avoid_topics: List[str] = Field(
        default_factory=list,
        description="Themen, die vermieden werden sollen",
    )


class VerticalRoutes(BaseModel):
    """Route-Konfiguration."""

    model_config = ConfigDict(protected_namespaces=())

    hidden: List[str] = Field(
        default_factory=list,
        description="Routes, die ausgeblendet werden sollen",
    )
    priority: List[str] = Field(
        default_factory=list,
        description="Routes, die priorisiert werden sollen",
    )
    custom_labels: Optional[Dict[str, str]] = Field(
        default=None,
        description="Custom Labels für Routes",
    )


class VerticalConfig(BaseModel):
    """Vollständige Vertical-Config."""

    model_config = ConfigDict(protected_namespaces=())

    features: VerticalFeatures
    terminology: VerticalTerminology
    ai_context: VerticalAIContext
    routes: VerticalRoutes


class Vertical(BaseModel):
    """Vertical-Metadaten."""

    model_config = ConfigDict(protected_namespaces=())

    id: str
    key: str = Field(..., description="z.B. 'mlm', 'real_estate', 'finance'")
    name: str = Field(..., description="Anzeigename")
    description: Optional[str] = None
    config: VerticalConfig
    created_at: Optional[str] = None
    updated_at: Optional[str] = None


# Default MLM Config (Fallback)
DEFAULT_MLM_CONFIG = VerticalConfig(
    features=VerticalFeatures(
        crm=True,
        finance=True,
        gamification=True,
        team=True,
        analytics=True,
        ai_coach=True,
        lead_hunter=True,
        follow_ups=True,
        templates=True,
        autopilot=True,
        cold_call=True,
        closing_coach=True,
        performance_insights=True,
        commission_tracker=True,
        genealogy=True,
        power_hour=True,
        churn_radar=True,
        roleplay_dojo=True,
        network_graph=True,
        field_ops=False,
        route_planner=False,
    ),
    terminology=VerticalTerminology(
        lead="Lead",
        deal="Einschreiben",
        revenue="Umsatz",
        commission="Provision",
        prospect="Interessent",
        customer="Partner",
        contact="Kontakt",
        pipeline="Pipeline",
        closing="Abschluss",
        sales="Verkauf",
        team="Team",
        downline="Downline",
        partner="Partner",
        enrollment="Einschreibung",
        signup="Anmeldung",
    ),
    ai_context=VerticalAIContext(
        persona="Du bist ein erfahrener Network Marketing Coach, der motivierend und unterstützend kommuniziert.",
        focus_topics=[
            "Network Marketing",
            "Team-Aufbau",
            "Downline-Management",
            "Provisionen",
            "Motivation",
            "Einschreibungen",
        ],
        industry_terms=[
            "Einschreibung",
            "Downline",
            "Upline",
            "Provision",
            "PV",
            "GV",
            "Partner",
            "Team",
        ],
        tone="motivierend",
        examples=[
            "Wie kann ich mein Team besser motivieren?",
            "Was sind die besten Strategien für Einschreibungen?",
        ],
        avoid_topics=["Pyramid Scheme", "Betrug"],
    ),
    routes=VerticalRoutes(
        hidden=[],
        priority=["/dashboard", "/leads", "/team", "/genealogy"],
    ),
)

